interceptors.go

```diff
@@ -0,0 +1,43 @@
+package sarama
+
+// ProducerInterceptor allows you to intercept (and possibly mutate) the records
+// received by the producer before they are published to the Kafka cluster.
+// https://cwiki.apache.org/confluence/display/KAFKA/KIP-42%3A+Add+Producer+and+Consumer+Interceptors#KIP42:AddProducerandConsumerInterceptors-Motivation
+type ProducerInterceptor interface {
+
+	// onSend is called when the producer message is intercepted. Please avoid
+	// modifying the message until it's safe to do so, as this is _not_ a copy
+	// of the message.
+	onSend(*ProducerMessage)
```

Comment:
These interfaces for the interceptors have only unexported methods. This means code outside the sarama package would be  unable to create types that satisfy the interface?

Reply:
🦅 👁️
yes, intention is to be used outside Sarama, one more reason to not use the package `main` on the tests, as this wasn't highlighted by the test.

I'll export the methods!
Thanks!